GetMessage/userPreferenceChecker/__tests__/userPreferencesCheckerFactory.test.ts

```diff
@@ -138,6 +139,24 @@ describe("userPreferencesCheckerFactory |> userPreferencesCheckerFactory", () =>
     expect(res).toStrictEqual(E.right(false));
   });
 
+  it("should call userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth if profile servicePreferencesSettings is of type LEGACY (version is -1)", async () => {
+    mockServicePreferencesGetter.mockReturnValueOnce(TE.of(O.none));
+
+    const res = await userPreferencesCheckerFactory(
+      {
+        ...aRetrievedProfile,
+        lastAppVersion: "UNKNOWN"
+      },
+      mockServicePreferencesGetter,
+      MIN_READ_STATUS_PREFERENCES_VERSION
+    ).canAccessMessageReadStatus(aServiceId, aFiscalCode)();
```

Comment:
This test is correct but here we are not testing the exact scenario that we want to, I would use a `MIN_READ_STATUS_PREFERENCES_VERSION` that would not make `isAppVersionHandlingReadAuthorization` pass. In fact this test would pass even if the `servicePreferenceSettings.version` would be a `NonNegativeInteger`.
This is what i suggest: 
```suggestion
  it("should call userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth if profile servicePreferencesSettings is of type LEGACY (version is -1)", async () => {
    mockServicePreferencesGetter.mockReturnValueOnce(TE.of(O.none));

    const res = await userPreferencesCheckerFactory(
      {
        ...aRetrievedProfile,
        lastAppVersion: "UNKNOWN"
      },
      mockServicePreferencesGetter,
      1.0.0 as Semver
    ).canAccessMessageReadStatus(aServiceId, aFiscalCode)();
```

Reply:
You found a problem, but the analysis was wrong. Legacy should go "unknown" when app version is not unknown.
I fixed with
```
{
        ...aRetrievedProfile,
        lastAppVersion: MIN_READ_STATUS_PREFERENCES_VERSION
      },
```
Now if you send a non negative integer this will not pass.